projects/ui/src/state/beanstalk/field/morning.ts

```diff
@@ -0,0 +1,145 @@
+import { useDispatch } from 'react-redux';
+import { useCallback, useEffect } from 'react';
+import { useBeanstalkContract } from '~/hooks/ledger/useContract';
+import { useAppSelector } from '~/state';
+
+import { useFetchBeanstalkField } from './updater';
+import { BEAN } from '~/constants/tokens';
+import { tokenResult } from '~/util';
+import { getNowRounded, getDiffNow } from '~/state/beanstalk/sun';
+import { updateTotalSoil } from './actions';
+import { useSeasonalTemperatureQuery } from '~/generated/graphql';
+
+/**
+ * useUpdateMorningField's primary function is to ensure that the redux store data
+ * reflects the most on-chain data during & right after the morning.
+ */
+
+export function useUpdateMorningField() {
+  /// App State
+  const morning = useAppSelector((s) => s._beanstalk.sun.morning);
+  const season = useAppSelector((s) => s._beanstalk.sun.season);
+  const soil = useAppSelector((s) => s._beanstalk.field.soil);
+  const temperature = useAppSelector((s) => s._beanstalk.field.temperature);
+  const next = useAppSelector((s) => s._beanstalk.sun.morningTime.next);
+
+  /// Fetch
+  const [fetchBeanstalkField] = useFetchBeanstalkField();
+  const temperatureQuery = useSeasonalTemperatureQuery({
+    fetchPolicy: 'cache-and-network',
+    skip: morning.isMorning,
+  });
+
+  /// Contract
+  const beanstalk = useBeanstalkContract();
+
+  const dispatch = useDispatch();
+
+  /// Derived
+  const morningBlock = morning.blockNumber;
+  const sunriseBlock = season.sunriseBlock;
+  const deltaBlocks = morningBlock.minus(sunriseBlock);
+
+  /// Temperature Season Data
+  const seasonData = temperatureQuery.data?.seasons;
+  const currentSeason = season.current;
+  const first = seasonData?.[0];
+
+  /// -------------------------------------
+  /// Callbacks
+
+  const fetchSoil = useCallback(async () => {
+    if (!beanstalk) {
+      console.debug(`[beanstalk/field/morning] fetch: contract undefined`);
+      return;
+    }
+    try {
+      const _soil = await beanstalk.totalSoil().then(tokenResult(BEAN));
+      console.debug('[beanstalk/field/morning] fetch: soil', _soil.toNumber());
+      if (!soil.eq(_soil)) {
+        dispatch(updateTotalSoil(_soil));
+      }
+    } catch (err) {
+      console.debug('[beanstalk/field/morning] fetch FAILED', err);
+    }
+  }, [soil, beanstalk, dispatch]);
+
+  /// -------------------------------------
+  /// Effects
+
+  // If it is morning, then we fetch the soil every 4 seconds
+  useEffect(() => {
+    if (!morning.isMorning) return;
+
+    const soilUpdateInterval = setInterval(() => {
+      const now = getNowRounded();
+
+      const remaining = getDiffNow(next, now).as('seconds');
+      if (remaining % 4 === 0) {
```

Comment:
Let's move the `4` to a constant at the top of the file and convert to a millisecond value, something like `MORNING_SOIL_REFRESH_RATE_MS`.